src/components/shared/ManagedTree.js

```diff
@@ -68,14 +69,36 @@ class ManagedTree extends Component {
     }
   }
 
-  setExpanded = (item: Item, isExpanded: boolean) => {
+  setExpanded = (item: Item, isExpanded: boolean, isRecursive: boolean) => {
     const expanded = this.state.expanded;
-    const itemPath = this.props.getPath(item);
+    let itemPath = this.props.getPath(item);
     if (isExpanded) {
       expanded.add(itemPath);
     } else {
       expanded.delete(itemPath);
     }
+    if (isRecursive) {
+      let children = null;
+      let parents = [item];
+      while (parents.length) {
+        children = [];
+        for (const parent of parents) {
+          if (!nodeHasChildren(parent)) {
+            continue;
+          }
+          for (const child of parent.contents) {
+            itemPath = this.props.getPath(child);
+            if (isExpanded) {
+              expanded.add(itemPath);
+            } else {
+              expanded.delete(itemPath);
+            }
+            children.push(child);
+          }
+        }
```

Comment:
This for loop can become
```js
        for (const parent of parents) {
          if (nodeHasChildren(parent)) {
            for (const child of parent.contents) {
              toggleItem(child);
              children.push(child);
            }
          }
        }
```

Reply:
We should not need the `continue` ...

Reply:
And may not need `nodeHasChildren` since we only use `parent.contents` anyway, we can just check that